### PR TITLE
feat: react query client

### PR DESCRIPTION
**Pull request by Pagebakers**
#### Checklist

- [x] run `make test`
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by StarpTech — APPROVED** [review]
Awesome work!

File: packages/react-query/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
```
[the remark this answers is not in the document]
Comment (StarpTech) [reply packages/react-query/CHANGELOG.md:3]: ok, let's see

**Review by thisisnithin — APPROVED** [review]

**Review by StarpTech — CHANGES_REQUESTED** [review]
Any reason why you didn't add type tests?

**Comment by Pagebakers** [comment]
> Any reason why you didn't add type tests?

No specific reason, I'll add them.

**Comment by Pagebakers** [comment]
@StarpTech Updated tests and added type tests to the package.

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**